Jenkinsfile

```
@@ -146,6 +146,39 @@ pipeline {
             }
           }
         }
+        stage('Publish snapshot packages') {
+          options { skipDefaultCheckout() }
+          environment {
+            PATH = "${env.WORKSPACE}/.local/bin:${env.WORKSPACE}/bin:${env.PATH}"
+            DOCKER_REGISTRY = 'docker.elastic.co'
+            DOCKER_REGISTRY_SECRET = 'secret/observability-team/ci/docker-registry/prod'
+            GCS_ACCOUNT_SECRET = 'secret/observability-team/ci/snapshoty'
+          }
+          // when { branch 'main' }
+          steps {
+            withGithubNotify(context: 'Publish snapshot packages') {
+              deleteDir()
+              unstash 'source'
+              unstash 'packages'
+              dockerLogin(secret: env.DOCKER_REGISTRY_SECRET, registry: env.DOCKER_REGISTRY)
+              script {
+                withSecretVault(
+                  secret: env.GCS_ACCOUNT_SECRET,
+                  data: [
+                    'client_email': 'GCS_CLIENT_EMAIL',
+                    'private_key': 'GCS_PRIVATE_KEY',
+                    'private_key_id': 'GCS_PRIVATE_KEY_ID',
+                    'project_id': 'GCS_PROJECT'
+                  ]
+                ) {
+                  docker.image('docker.elastic.co/observability-ci/snapshoty:v1').inside() {
+                    sh(script: "cd ${BASE_DIR} && snapshoty --config ./.ci/snapshoty.yml upload --bucket-name obtl-artifacts")
+                  }
+                }
+              }
```

If https://github.com/elastic/apm-pipeline-library/pull/1961 then

```suggestion
              dir(BASE_DIR) {
                snapshoty(dockerSecret: env.DOCKER_REGISTRY_SECRET, dockerRegistry: env.DOCKER_REGISTRY,
                                   bucket: 'obtl-artifacts' , gcsAccountSecret: env.GCS_ACCOUNT_SECRET)
              }
```